wrappers/desktop.go

```diff
@@ -318,8 +416,31 @@ func RemoveSnapDesktopFiles(s *snap.Info) error {
 		return nil
 	}
 
-	desktopFilesGlob := fmt.Sprintf("%s_*.desktop", s.DesktopPrefix())
-	_, removed, err := osutil.EnsureDirState(dirs.SnapDesktopFilesDir, desktopFilesGlob, nil)
+	installedDesktopFiles, err := findDesktopFiles(dirs.SnapDesktopFilesDir)
+	if err != nil {
+		return err
+	}
+
+	desktopFilesGlobs := []string{fmt.Sprintf("%s_*.desktop", s.DesktopPrefix())}
+	for _, desktopFile := range installedDesktopFiles {
+		instanceName, err := readSnapInstanceName(desktopFile)
+		if err != nil {
+			// cannot read instance name from desktop file, ignore
+			logger.Noticef("cannot read instance name: %s", err)
+			continue
+		}
+		base := filepath.Base(desktopFile)
```

Comment:
and this can share code by having a  `forAllDesktopFiles(cb func(base, instaneName) error) ferror` unction that implements the looping?